File: OrchidCore/src/main/java/com/eden/orchid/api/server/admin/AdminList.java
package com.eden.orchid.api.server.admin;

import com.eden.orchid.api.options.Descriptive;

import java.util.Collection;

/**
 * Admin Lists add collections of items to be displayed within the admin area for the purposes of self-documentation and
 * discovery.
 *
 * @since v1.0.0
 * @orchidApi extensible
 */
public interface AdminList extends Descriptive {

    String getKey();

    Class<?> getListClass();

    Collection<Class<?>> getItems();

    boolean isImportantType();

    default String getDescriptiveName() {
        return Descriptive.getDescriptiveName(getListClass());
    }

    default String getDescription() {
        return Descriptive.getDescription(getListClass());
    }
}

File: orchid-core/src/main/kotlin/com/eden/orchid/api/server/admin/AdminList.kt
package com.eden.orchid.api.server.admin

import com.eden.orchid.api.options.Descriptive

/**
 * Admin Lists add collections of items to be displayed within the admin area for the purposes of self-documentation and
 * discovery.
 */
interface AdminList : Descriptive {
    val key: String
    val listClass: Class<*>
    val items: Collection<Class<*>?>
    val isImportantType: Boolean

    @JvmDefault
    override fun getDescriptiveName(): String {
        return Descriptive.getDescriptiveName(listClass)
    }

    @JvmDefault
    override fun getDescription(): String {
        return Descriptive.getDescription(listClass)
    }
}
